Update README.md for dataservice-write. (#357)
heremaps/here-data-sdk-typescript: @here/olp-sdk-dataservice-write/README.md

# DataService Write

## Overview

> This repository contains the complete source code for the HERE Data SDK for TypeScript Dataservice Write `@here/olp-sdk-dataservice-write` project. `olp-sdk-dataservice-write` is a TypeScript library that will upload data to the HERE platform catalogs.

Reviewer: ```suggestion
This repository contains the complete source code for the HERE Data SDK for TypeScript Dataservice Write (`@here/olp-sdk-dataservice-write`) project. `olp-sdk-dataservice-write` is a TypeScript library that you can use to upload data to the HERE platform catalogs.
```

> # Publish data

Reviewer: ```suggestion
## Publish data
```

> You can use the HERE Data SDK for TypeScript to publish data to catalogs.

Reviewer: Delete

Now, we only support publishing data to a versioned layer.

You can publish data to a [versioned layer](https://developer.here.com/olp/documentation/data-user-guide/portal/layers/layers.html#versioned-layers), if you need to store and access the history of previous data updates. To achieve consistency between layers, publish any update that affects multiple versioned layers together in one publication. You can access a new catalog version when all layers are updated, and the publication is finalized. Once you publish a version, you cannot change the data in that version and can [remove](#cancel) it only by removing the whole catalog version.

The maximum number of partitions you can upload and publish in one request is 1,000. If you have more than 1,000 partitions, upload the data and metadata for the first 1,000 partitions, and then do the same for the next set of partitions.

**To publish data to the versioned layer:**

1. Import data from the `olp-sdk-dataservice-write` module.

    Add to your dependencies:

    ````
    "@here/olp-sdk-dataservice-api": "1.5.0",
    "@here/olp-sdk-dataservice-write": "1.0.0"

>     ```

Reviewer: Make it step 1.

```
1. Add the `@here/olp-sdk-dataservice-api` and `@here/olp-sdk-dataservice-write` to your dependencies.
    ````
    "@here/olp-sdk-dataservice-api": "1.5.0",
    "@here/olp-sdk-dataservice-write": "1.0.0"
    ```
2. Import data from the `olp-sdk-dataservice-write` module.
    ```typescript
       import {
         VersionedLayerClient,
         StartBatchRequest,
         CompleteBatchRequest,
         UploadPartitionsRequest,
         UploadBlobRequest,
       } from "@here/olp-sdk-dataservice-write";
       ```
```

To get bundled files with a source map, run:

```sh
npm run bundle
```

To get minified version for production, run:

```sh
npm run bundle:prod
```

To get bundled and minified JavaScript files, run:

```sh
npm run prepublish-bundle
```

## Publish data

Now, we only support publishing data to a versioned layer.

You can publish data to a [versioned layer](https://developer.here.com/olp/documentation/data-user-guide/portal/layers/layers.html#versioned-layers), if you need to store and access the history of previous data updates. To achieve consistency between layers, publish any update that affects multiple versioned layers together in one publication. You can access a new catalog version when all layers are updated, and the publication is finalized. Once you publish a version, you cannot change the data in that version and can [remove](#cancel) it only by removing the whole catalog version.

The maximum number of partitions you can upload and publish in one request is 1,000. If you have more than 1,000 partitions, upload the data and metadata for the first 1,000 partitions, and then do the same for the next set of partitions.

**To publish data to the versioned layer:**

> 1. Add to your dependencies:

Reviewer: ```suggestion
1. Add the `@here/olp-sdk-dataservice-api` and `@here/olp-sdk-dataservice-write` packages to your dependencies.
```

    ```
    "@here/olp-sdk-dataservice-api": "1.5.0",
    "@here/olp-sdk-dataservice-write": "1.0.0"
    ```

2. Import data from the `olp-sdk-dataservice-write` module.

>     And import classes:

Reviewer: Delete

## Directory Layout

Here is an overview of the top-level files of the repository:

    |
    +- @here/olp-sdk-dataservice-write
                        |
                        +- lib    # Implementation of the project
                        |
                        +- test   # Test code

## Development

### Prerequisites

The following NPM packages are required to build/test the library:

    - node: >= 10.0.0
    - npm: >= 6.0.0

### Build

Open a command prompt of the working tree's root directory and type:

```sh
npm install
npm run build
```

### Test

Open a command prompt of the working tree's root directory and type:

```sh
npm run test
```

### Generate a Bundle

If you want to have a compiled project, you can use bundle commands. After running each of the following commands in the `@here/olp-sdk-dataservice-read` folder from the root folder, you get the JavaScript bundled files.

To get bundled files with a source map, run:

```sh
npm run bundle
```

To get minified version for production, run:

```sh
npm run bundle:prod
```

To get bundled and minified JavaScript files, run:

```sh
npm run prepublish-bundle

> 

Reviewer: Put it before "# Publish data"
